internal/handlers/common/update.go

```diff
@@ -26,16 +26,18 @@ import (
 // UpdateDocument updates the given document with a series of update operators.
 // Returns true if document was changed.
 func UpdateDocument(doc, update *types.Document) (bool, error) {
+	var changed bool
 	for _, updateOp := range update.Keys() {
 		updateV := must.NotFail(update.Get(updateOp))
 
 		switch updateOp {
-		case "$set":
+		case "$set",
+			"$setOnInsert":
 
 			switch setDoc := updateV.(type) {
 			case *types.Document:
 				if setDoc.Len() == 0 {
-					return false, nil
+					continue
```

Comment:
If I'm not mistaken, that changes the logic for the `$set` case. Why before it was ok to just return the result, and now it's not ok anymore, and we need to go to the next iteration?

Reply:
> If I'm not mistaken, that changes the logic for the `$set` case. Why before it was ok to just return the result, and now it's not ok anymore, and we need to go to the next iteration?

> This formatting looks a bit unusual to me. But I'm not sure I know the right formatting :) Why are both conditions not on the same line? Is it something linter-specific?

a new test case was added:
```go
                "set-setoninsert": {
			filter: bson.D{{"_id", "test"}},
			query: bson.D{
				{"$set", bson.D{{"foo", int32(12)}}},
				{"$setOnInsert", bson.D{{"value", math.NaN()}}},
			},
			res: bson.D{{"_id", "test"}, {"foo", int32(12)}, {"value", math.NaN()}},
		},
```
and it was found that when multiply changes in $set clause then `modified` count gives the wrong number.

Reply:
Let's add a test for `$set` that shows the problem in the previous code